src/SuperMint.sol

```diff
@@ -7,11 +7,15 @@ import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
 contract SuperMint is IERC721Receiver {
   MaxMint721 implementation;
   uint maxMints;
-  bool complete;
+  //added a new variable to perform the iteration and removed the bool variable
+  uint iterator;
```

Comment:
instead of using a separate variable here for the iterator we can just update `maxMints`